concourse/pipelines/templates/dev_build_pipeline-tpl.yml

```
@@ -369,228 +369,116 @@ resources:
-  - task: test_pxf-gp[[gp_ver]]-hdp2_on_ubuntu18
-    file: pxf_src/concourse/tasks/test.yml
-    image: gpdb[[gp_ver]]-pxf-dev-ubuntu18-image
-    params:
-      ACCESS_KEY_ID: ((tf-machine-access-key-id))
-      GP_VER: [[gp_ver]]
-      GROUP: gpdb,proxy,profile
-      PXF_BASE_DIR: /home/gpadmin/pxf
-      SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
-{% if slack_notification %}
-    <<: *slack_alert
-{% endif %}
-{% set gp_ver = None %}
-
-## ---------- RHEL 8 Swimlane (uses Rocky 8 images) ----------
-{% for gp_ver in range(6, 8) %}
-- name: build_pxf-gp[[gp_ver]]_on_rhel8
-  plan:
-  - in_parallel:
-    - get: pxf_src
-      trigger: true
-    - get: gpdb_package
-      resource: gpdb[[gp_ver]]_rhel8_rpm_latest-0
-    - get: gpdb[[gp_ver]]-pxf-dev-rocky8-image
-    - get: pxf-build-dependencies
-  - task: build_pxf-gp[[gp_ver]]_on_rhel8
-    image: gpdb[[gp_ver]]-pxf-dev-rocky8-image
-    file: pxf_src/concourse/tasks/build.yml
-    params:
-      LICENSE: ((ud/pxf/common/rpm-license))
-      VENDOR: ((ud/pxf/common/rpm-vendor))
-  - put: pxf_gp[[gp_ver]]_tarball_rhel8
-    params:
-      file: dist/pxf-gp[[gp_ver]]-*.el8.tar.gz
-{% if gp_ver == 6 %}
-  - put: pxf_fdw_gp[[gp_ver]]_tarball_rhel8
-    params:
-      file: dist/pxf-fdw-gp[[gp_ver]]-*.el8.tar.gz
-{% endif %}
-
-- name: test_pxf-gp[[gp_ver]]-hdp2_on_rhel8
-  plan:
-  - in_parallel:
-    - get: pxf_src
-      passed: [build_pxf-gp[[gp_ver]]_on_rhel8]
-      trigger: true
-    - get: pxf_tarball
-      resource: pxf_gp[[gp_ver]]_tarball_rhel8
-      passed: [build_pxf-gp[[gp_ver]]_on_rhel8]
-    - get: gpdb_package
-      resource: gpdb[[gp_ver]]_rhel8_rpm_latest-0
-      passed: [build_pxf-gp[[gp_ver]]_on_rhel8]
-    - get: gpdb[[gp_ver]]-pxf-dev-rocky8-image
-      passed: [build_pxf-gp[[gp_ver]]_on_rhel8]
-    - get: pxf-automation-dependencies
-    - get: singlecluster
-      resource: singlecluster-hdp2
-  - task: test_pxf-gp[[gp_ver]]-hdp2_on_rhel8
-    file: pxf_src/concourse/tasks/test.yml
-    image: gpdb[[gp_ver]]-pxf-dev-rocky8-image
-    params:
-      ACCESS_KEY_ID: ((tf-machine-access-key-id))
-      GP_VER: [[gp_ver]]
-      GROUP: gpdb,proxy,profile
-      SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
+## ======================================================================
+##                    Build and Sanity Test Swimlanes
+## ----------------------------------------------------------------------
+##
+## The creation of the PXF Build and Sanity test jobs are driven by
+## the following platform_gp_combinations dictionary definition:
+##
+##   gp_ver: GPDB Major Version
+##   execution_os: OS platform in which build and test jobs are executed
+##   os_ver: OS Major version
+##   build_os: OS Platform in which the build was performed
+##
+## ----------------------------------------------------------------------
+{% set platform_gp_combinations = [
+    {'gp_ver': '5', 'execution_os': 'centos', 'os_ver': '7',  'build_os': 'centos'},
```

```suggestion
    {'gp_ver': '5', 'build_os': 'centos', 'test_os': 'centos', 'os_ver': '7'},
```
can we have this logical order ?